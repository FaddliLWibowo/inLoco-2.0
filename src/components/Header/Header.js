import React from 'react';

const Header = ({onHeaderClick}) => {
    return (
        <div className="application-header">
            <div className="application-header--logo-placeholder">
                <img src={require('../../assets/img/logo.png')} alt="InLoco" className="application-header--logo"/>
            </div>
            <a className="application-header--menu-button fa fa-bars" role="button" onClick={onHeaderClick}></a>
        </div>
    );
};

export default Header;
