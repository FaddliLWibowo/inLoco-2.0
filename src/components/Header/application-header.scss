.application-header {
    left: 15px;
    position: absolute;
    top: 15px;
    z-index: 1001;

    &--logo-placeholder {
        background-color: $logo-placeholder-background-color;
        border-radius: 4px;
        height: 58px;
        width: 230px;
        box-shadow: 0px 3px 5px 0px rgba(0, 0, 0, 0.72);
    }

    &--logo {
        height: 39px;
        margin: 8px 22px;
        width: 185px;
    }

    &--menu-button {
        background-color: $menu-bg-color;
        border-radius: 50%;
        color: $menu-text-color;
        cursor: pointer;
        display: block;
        font-size: 32px;
        height: 55px;
        margin-top: 25px;
        padding-top: 10px;
        text-align: center;
        text-decoration: none;
        width: 55px;
        box-shadow: 0px 3px 5px 0px rgba(0, 0, 0, 0.72);
        transition: background-color ease .15s;

        &:hover{
            background-color: $menu-bg-color-hover;
        }

    }
}
