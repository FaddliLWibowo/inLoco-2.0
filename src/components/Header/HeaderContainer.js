import React from 'react';
import Header from './Header';
import { connect } from 'react-redux';
import { showMenuLayer } from '../../actions/actions.js';

const mapDispatchToProps = (dispatch) => {
    return {
        onHeaderClick: () => {
            dispatch(showMenuLayer());
        }
    };
};

const HeaderContainer = connect(
    null,
    mapDispatchToProps
)(Header)

export default HeaderContainer;
