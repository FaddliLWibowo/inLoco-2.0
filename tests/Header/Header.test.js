import React from 'react';
import Header from '../../src/components/Header/Header.js';

it('Header component renders correctly with no data', () => {
    const component = shallow(<Header/>);
    expect(component).toMatchSnapshot();
});
