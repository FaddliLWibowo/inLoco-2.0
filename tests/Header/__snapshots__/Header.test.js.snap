// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly with no data 1`] = `
<div
  className="application-header"
>
  <div
    className="application-header--logo-placeholder"
  >
    <a
      href="http://mpemmapas.mprj.mp.br/home.html"
      target="_blank"
    >
      <img
        alt="InLoco"
        className="application-header--logo"
        src={Object {}}
      />
    </a>
  </div>
  <a
    className="application-header--menu-button fa fa-bars focus-menu"
    data-html2canvas-ignore={true}
    role="button"
  >
    <span
      className="menu-button--tootltip hidden"
    >
      Comece por aqui.
    </span>
  </a>
</div>
`;
